narwhals/_pandas_like/dataframe.py

```diff
@@ -249,8 +249,8 @@ def with_columns(
     def rename(self, mapping: dict[str, str]) -> Self:
         return self._from_dataframe(self._dataframe.rename(columns=mapping))
 
-    def drop(self, columns: str | Iterable[str]) -> Self:
-        return self._from_dataframe(self._dataframe.drop(columns=flatten(columns)))
+    def drop(self, *columns: str | Iterable[str]) -> Self:
+        return self._from_dataframe(self._dataframe.drop(columns=[*flatten(columns)]))
```

Comment:
i think it's just that pandas does some strange multi-index things with tuples, which we don't support anyway
```suggestion
        return self._from_dataframe(self._dataframe.drop(columns=list(flatten(columns)))
```